src/config/configuration.js

```diff
@@ -0,0 +1,57 @@
+export default class Configuration {
+  #tree
+  #listeners = []
+
+  constructor(initial = {}) {
+    this.#tree = initial
+  }
+
+  // returns bools early, null if key is missing
+  get(path = "") {
+    let node = this.#tree
+    const keys = path.split(".")
+    for (const key of keys) {
+      if (!Object.hasOwn(node, key)) {
+        return null
+      }
+
+      node = node[key]
+      if (node === true || node === false) {
+        return node
+      }
+    }
+
+    return node
+  }
+
+  merge(config, into = this.#tree) {
```

Comment:
I think the `into` param is a bit confusing. Why would you like to change that to other thing? This is an instance method, it is expected it will act on its private data. I would remove.